src/host/screenInfo.cpp

```diff
@@ -1423,22 +1423,51 @@ bool SCREEN_INFORMATION::IsMaximizedY() const
     }
 
     // Save cursor's relative height versus the viewport
-    SHORT const sCursorHeightInViewportBefore = _textBuffer->GetCursor().GetPosition().Y - _viewport.Top();
+    short cursorHeightInViewportBefore = _textBuffer->GetCursor().GetPosition().Y - _viewport.Top();
 
+    auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
+    const bool isConpty = gci.IsInVtIoMode();
     HRESULT hr = TextBuffer::Reflow(*_textBuffer.get(), *newTextBuffer.get());
+    const bool widthChanged = coordNewScreenSize.X != _textBuffer->GetSize().Width();
 
     if (SUCCEEDED(hr))
     {
         Cursor& newCursor = newTextBuffer->GetCursor();
         // Adjust the viewport so the cursor doesn't wildly fly off up or down.
-        SHORT const sCursorHeightInViewportAfter = newCursor.GetPosition().Y - _viewport.Top();
+        const short cursorHeightInViewportAfter = newCursor.GetPosition().Y - _viewport.Top();
         COORD coordCursorHeightDiff = { 0 };
-        coordCursorHeightDiff.Y = sCursorHeightInViewportAfter - sCursorHeightInViewportBefore;
+        coordCursorHeightDiff.Y = cursorHeightInViewportAfter - cursorHeightInViewportBefore;
         LOG_IF_FAILED(SetViewportOrigin(false, coordCursorHeightDiff, true));
 
         _textBuffer.swap(newTextBuffer);
     }
 
+    // GH#3490 - In conpty mode, We want to invalidate all of the viewport that
+    // might have been below any wrapped lines, up until the last character of
+    // the buffer. Lines that were wrapped may have been re-wrapped during this
+    // resize, so we want them repainted to the terminal. We don't want to just
+    // invalidate everything though - if there were blank lines at the bottom,
+    // those can just be ignored.
+    if (isConpty && widthChanged)
+    {
+        // Loop through all the rows of the old buffer and reprint them into the new buffer
+        const auto bottom = std::max(_textBuffer->GetCursor().GetPosition().Y,
+                                     std::min(_viewport.BottomInclusive(),
+                                              _textBuffer->GetLastNonSpaceCharacter().Y));
+        bool foundWrappedLine = false;
+        for (short y = _viewport.Top(); y <= bottom; y++)
+        {
+            // Fetch the row and its "right" which is the last printable character.
+            const ROW& row = _textBuffer->GetRowByOffset(y);
+            const CharRow& charRow = row.GetCharRow();
+            if (foundWrappedLine || charRow.WasWrapForced())
+            {
+                foundWrappedLine = true;
+                _renderTarget.TriggerRedraw(Viewport::FromDimensions({ 0, y }, _viewport.Width(), 1));
+            }
+        }
+    }
+

```

Comment:
Nnnnnn. I don't really like this but I don't have another offering at hand so maybe it has to stick for the short term.

I can see this going where the `Resize` method on the `TextBuffer` becomes more responsible for providing appropriate viewport notifications since it seems like every caller needs to figure out something based on the `Cursor.GetPosition()` after the fact. And then to add some... intricate rendering details about the `VtRenderer` to this side of the equation feels dirty, design wise. I felt like determining what was different fit better in the renderer side.

But I don't know what to do about it right now. I'm guessing there just isn't enough data available to the `VtRenderer` to know that this happened? What specifically was missing? Isn't the width available to the `VtRenderer` via the `RenderData` interface? Or is it that it can't reach that specifically because `RenderData` is only in the base renderer and it can't query that?

Reply:
Okay so this is the part that actually might have messed up `gnome-terminal` and conhost/inception. 

The real trick here is that we _don't_ want to just invalidate the blank lines at the bottom of the buffer. However, I'm afarid now that this might _never_ work quite right, if the terminal resized again before this frame is printed. I'm still investigating if this PR is the right solutions - it might not turn out that way.